lib/Dialect/Secret/IR/SecretPatterns.cpp

```diff
@@ -178,6 +181,138 @@ LogicalResult CaptureAmbientScope::matchAndRewrite(
   return success();
 }
 
+LogicalResult MergeAdjacentGenerics::matchAndRewrite(
+    GenericOp genericOp, PatternRewriter &rewriter) const {
+  GenericOp nextGenericOp = dyn_cast<GenericOp>(genericOp->getNextNode());
+  if (!nextGenericOp) {
+    return failure();
+  }
+
+  SmallVector<Type> newResultTypes;
+  newResultTypes.reserve(genericOp->getNumResults() +
+                         nextGenericOp->getNumResults());
+  newResultTypes.append(genericOp->getResultTypes().begin(),
+                        genericOp->getResultTypes().end());
```

Comment:
In the case where the yielded value of the first is only used as an argument to the second, we will still end up yielding it, it seems, right? Even though it may not have any more uses after this (does canonicalize clean up unused yielded values? i feel like that was a thing)

Reply:
Yes, RemoveUnusedYieldedValues: https://github.com/google/heir/blob/ea8e2f305657e40ce2c14b6d9ed76fadcc0cbc27/include/Dialect/Secret/IR/SecretPatterns.h#L92 which is part of canonicalization: https://github.com/google/heir/blob/ea8e2f305657e40ce2c14b6d9ed76fadcc0cbc27/lib/Dialect/Secret/IR/SecretOps.cpp#L334